### ISSUE TITLE
[Bug] [Android] RefreshView is broken when CollectionView contains single/"zero height" element in GroupHeader

### ISSUE THREAD

**Issue by bondarenkod** [xamarin/Xamarin.Forms#12056]
### Description
Using an empty view inside CollectionView.GroupHeaderTemplate as a view prevent RefreshView from reacting on user interaction ("pull to refresh"). 

### Steps to Reproduce

1. Put into CollectionView.GroupHeaderTemplate single and empty element to display absent group header:
  ```
<CollectionView.GroupHeaderTemplate>
    <DataTemplate>
        <Grid />
    </DataTemplate>
</CollectionView.GroupHeaderTemplate>
```
2. Create a page with CollectionView that can accept groups and put it inside RefreshView, you can use default modern template for XF application with tabs from recent versions of VS2019. 
### Expected Behavior
RefreshView control should response on user interaction. 
### Actual Behavior
The user interaction is ignored by RefreshView.

### Basic Information

- Version with issue:
- Last known good version: I have no idea, but the bug is existing on 4.5+ versions at least. 
- Platform Target Frameworks: 
  - iOS:  NO
  - Android: YES
  - UWP: didn't check 

### Screenshots
![09_09_2020__00_45_27__582306ee-32f5-4b73-9b28-de23d314b583](https://user-images.githubusercontent.com/3184414/92530845-d17c6380-f235-11ea-85d2-6a0101a63b99.gif)

### Reproduction Link
https://github.com/bondarenkod/xf_collectionview_refreshview_pulltorefresh_issue 
see the files 
[/Views/ItemsPage.xaml](https://github.com/bondarenkod/xf_collectionview_refreshview_pulltorefresh_issue/blob/master/refreshview_emptygroup/refreshview_emptygroup/Views/ItemsPage.xaml)
[/Views/ItemsPage2.xaml](https://github.com/bondarenkod/xf_collectionview_refreshview_pulltorefresh_issue/blob/master/refreshview_emptygroup/refreshview_emptygroup/Views/ItemsPage2.xaml)

### Workaround
Put an addition element inside the root one
```
 <CollectionView.GroupHeaderTemplate>
    <DataTemplate>
        <Grid>
            <Grid />
        </Grid>
    </DataTemplate>
</CollectionView.GroupHeaderTemplate>
```

**Comment by StephaneDelcroix** [comment]
I don't think it prevents anything, but as it has an empty size, you can't slide it.

@hartez your opinion on this ?

**Comment by bondarenkod** [comment]
the same project, ios
![10_09_2020__16_27_05__a633cefb-037b-4442-9054-5fb254b91d06](https://user-images.githubusercontent.com/3184414/92735949-81f77e00-f382-11ea-8ab5-d7da907dcd24.gif)
